packages/kbn-apm-synthtrace/src/scenarios/aws_lambda.ts

```diff
@@ -26,27 +26,68 @@ const scenario: Scenario<ApmFields> = async (runOptions: RunOptions) => {
       const instance = apm.service('lambda-python', ENVIRONMENT, 'python').instance('instance');
 
       const traceEventsSetups = [
-        { functionName: 'lambda-python-1', coldStart: true },
-        { functionName: 'lambda-python-2', coldStart: false },
+        {
+          functionName: 'lambda-python-1',
+          coldStart: true,
+          billedDuration: 4000,
+          timeout: 10000,
+          coldStartDuration: 4000,
+          faasDuration: 4000,
+        },
+        {
+          functionName: 'lambda-python-2',
+          coldStart: false,
+          billedDuration: 4000,
+          timeout: 10000,
+          coldStartDuration: 0,
+          faasDuration: 4000,
+        },
       ];
 
-      const traceEvents = ({ functionName, coldStart }: typeof traceEventsSetups[0]) => {
-        return timestamps.generator((timestamp) =>
+      const traceEvents = ({
+        functionName,
+        coldStart,
+        billedDuration,
+        coldStartDuration,
+        faasDuration,
+        timeout,
+      }: typeof traceEventsSetups[0]) => {
+        const awsLambdaEvents = timestamps.generator((timestamp) =>
           instance
             .transaction('GET /order/{id}')
             .defaults({
               'service.runtime.name': 'AWS_Lambda_python3.8',
               'cloud.provider': 'aws',
               'cloud.service.name': 'lambda',
               'cloud.region': 'us-east-1',
-              'faas.id': `arn:aws:lambda:us-west-2:123456789012:function:${functionName}`,
-              'faas.coldstart': coldStart,
-              'faas.trigger.type': 'other',
             })
             .timestamp(timestamp)
             .duration(1000)
             .success()
         );
+
+        const metricsets = range
+          .interval('30s')
+          .rate(1)
+          .generator((timestamp) =>
+            instance
+              .appMetrics({
+                'system.memory.actual.free': 94371840, // ~0.08 gb
+                'system.memory.total': 536870912, // 0.5gb
+                'system.cpu.total.norm.pct': 0.6,
+                'system.process.cpu.total.norm.pct': 0.7,
+                'faas.id': `arn:aws:lambda:us-west-2:123456789012:function:${functionName}`,
+                'faas.coldstart': coldStart,
+                'faas.trigger.type': 'other',
```

Comment:
this is metadata, it should be set in the `instance`. you might want to create a sub-class of `instance` for a lambda, similar to `browser`.